frontend/packages/shared/src/components/GiteaHeader/VersionControlButtons/components/ShareChangesPopover/CommitAndPushContent/FileChangesInfoModal/FilePath/FilePathUtils.test.ts

```diff
@@ -0,0 +1,46 @@
+import { convertPureGitDiffToUserFriendlyDiff } from 'app-shared/components/GiteaHeader/VersionControlButtons/components/ShareChangesPopover/CommitAndPushContent/FileChangesInfoModal/FilePath/FilePathUtils';
+
+const diffStringMock = `diff --git a/fileName.json b/fileName.json
+index 0909a03..527e226 100644
+--- a/fileName.json
++++ b/fileName.json
+@@ -2,6 +2,30 @@
+- old line
++ new line
+unchanged line
+\ No newline at end of file`;
+
+describe('FilePathUtils', () => {
+  describe('convertPureGitDiffToUserFriendlyDiff', () => {
+    it('removes all metadata including line with double @ in the diff string', () => {
+      const userFriendlyDiff: string[] = convertPureGitDiffToUserFriendlyDiff(diffStringMock);
+
+      expect(userFriendlyDiff).toEqual(['- old line', '+ new line', 'unchanged line']);
+    });
+
+    it('removes last metadata line if it exists', () => {
+      const userFriendlyDiff: string[] = convertPureGitDiffToUserFriendlyDiff(diffStringMock);
+
+      expect(userFriendlyDiff).not.toContain('No newline at end of file');
+    });
+
+    it('returns last line if it is not metadata', () => {
+      diffStringMock.replace(' No newline at end of file', '');
+      const userFriendlyDiff: string[] = convertPureGitDiffToUserFriendlyDiff(diffStringMock);
+
+      expect(userFriendlyDiff).toEqual(['- old line', '+ new line', 'unchanged line']);
+    });
+
+    it('handles binary diff changes metadata', () => {
+      const binaryGitDiffString = `diff --git a/App/wwwroot/image.png b/App/wwwroot/image.png
+      new file mode 100644
+      index 0000000..48550ba
+      Binary files /dev/null and b/App/wwwroot/image.png differ`;
+      const userFriendlyDiff: string[] = convertPureGitDiffToUserFriendlyDiff(binaryGitDiffString);
+
+      expect(userFriendlyDiff).toEqual([
+        '      Binary files /dev/null and b/App/wwwroot/image.png differ',
```

Comment:
Vi bør nok vurdere et eget skjermbilde for disse situasjonene, men det har liten hensikt å ta fatt på det før vi vet mer om hvor mye dette brukes.

Reply:
Jeg har laget et issue for å hente det faktiske bildet. Om det var det du tenkte på?

Reply:
Ja, det er lurt. Men vi bør kanskje også kunne håndtere andre filtyper, som video og lyd, og kanskje også en slags "dummy-visning" for filer som ikke lar seg presentere på en fornuftig måte. Men vi trenger ikke ta tak i det nå.

Reply:
Tror video og lyd evt ville blitt løst på samme måte? Man henter jo bare filen uavhengig av hvordan binary-fil det er? Men jeg vet ikke helt hvordan det ser ut med andre type filer skal sies. Og ja, er nok uansett noe som burde løses i den andre saken 👍